tests/helm/metrics_test.go

```diff
@@ -53,3 +53,155 @@ metadata:
+  metrics:
+    config:
+      extraProcessors:
+        - filter/1:
+            metrics:
+              include:
+                match_type: regexp
+                metric_names:
+                  - receiver_mock_.*
+                resource_attributes:
+                  - Key: k8s.pod.name
+                    Value: app.*
+              exclude:
+                match_type: strict
+                metric_names:
+                  - receiver_mock_logs_count
+        - transform/rename_metric:
+            metric_statements:
+              - context: metric
+                statements:
+                  - set(name, "rrreceiver_mock_metrics_count") where name == "receiver_mock_metrics_count"
+        - transform/rename_metadata:
+            metric_statements:
+              - context: resource
+                statements:
+                  - set(attributes["k8s.pod.pod_name_new"], attributes["k8s.pod.pod_name"])
+                  - delete_key(attributes, "k8s.pod.pod_name")
+                  - set(attributes["my.static.value"], "<static_value>")
+`
+	otelConfigYaml := GetOtelConfigYaml(t, valuesYaml, templatePath)
+
+	var otelConfig struct {
+		Processors struct {
+			Filter struct {
+				Metrics struct {
+					Include struct {
+						MatchType string `yaml:"match_type"`
+					}
+					Exclude struct {
+						MatchType string `yaml:"match_type"`
+					}
+				}
+			} `yaml:"filter/1"`
+			RenameMetric struct {
+				MetricStatements []struct {
+					Context    string   `yaml:"context"`
+					Statements []string `yaml:"statements"`
+				} `yaml:"metric_statements"`
+			} `yaml:"transform/rename_metric"`
+			RenameMetadata struct {
+				MetricStatements []struct {
+					Context    string   `yaml:"context"`
+					Statements []string `yaml:"statements"`
+				} `yaml:"metric_statements"`
+			} `yaml:"transform/rename_metadata"`
+		}
+	}
+
+	err := yaml.Unmarshal([]byte(otelConfigYaml), &otelConfig)
+	require.NoError(t, err)
+
+	require.Equal(t, "regexp", otelConfig.Processors.Filter.Metrics.Include.MatchType)
+	require.Equal(t, "strict", otelConfig.Processors.Filter.Metrics.Exclude.MatchType)
+
+	renameMetricStatements := []string{
+		`set(name, "rrreceiver_mock_metrics_count") where name == "receiver_mock_metrics_count"`,
+	}
+	require.Equal(t, "metric", otelConfig.Processors.RenameMetric.MetricStatements[0].Context)
+	require.Equal(t, renameMetricStatements, otelConfig.Processors.RenameMetric.MetricStatements[0].Statements)
+
+	renameMetadatatatements := []string{
+		`set(attributes["k8s.pod.pod_name_new"], attributes["k8s.pod.pod_name"])`,
+		`delete_key(attributes, "k8s.pod.pod_name")`,
+		`set(attributes["my.static.value"], "<static_value>")`,
+	}
+	require.Equal(t, "resource", otelConfig.Processors.RenameMetadata.MetricStatements[0].Context)
+	require.Equal(t, renameMetadatatatements, otelConfig.Processors.RenameMetadata.MetricStatements[0].Statements)
+}
+
+func TestMetadataMetricsOtelConfigExtraProcessorsPipeline(t *testing.T) {
```

Comment:
These two tests use the same configuration, could we merge them into one? The actual assertions in this test aren't very complex, adding them to the first test should be ok.

I'd also consider moving the valuesYaml to a file in `testdata/`, it's a bit large for an inline value imo.

Reply:
done